README.md

```diff
@@ -208,7 +208,35 @@ At a minimum, all Roaring implementations should be able to parse the two files
 
 ## Unsigned integers in Java
 
-
 Java lacks native unsigned integers, but integers are still considered to be unsigned within Roaring, and ordered  according to  ``Integer.compareUnsigned``. 
 
 For integers in [0,2147483647], the unsigned and signed integers are undistinguisable since Java relies a 32-bit two's complement format for its ``int`` type.
+
+# Extention for 64B implementations
+
+Some Roaring bitmap implementations may offer a 64-bits implementation. This section proposes a portable format, compatible with some (but not all) 
+64-bits implementations. This format is naturally compatible with implementations based on a conventional red-black-tree (as the serialization format 
+is similar to the in-memory layout). The keys would be 32-bit integers representing the most significant 32~bits of elements whereas the values of the 
+tree are 32-bit Roaring bitmaps. The 32-bit Roaring bitmaps represent the least significant bits of a set of elements.
+
+## General layout
+
+All words are written using little endian encoding.
+
+- Write as long/uint64 the distinct number of buckets (restricted to integers in [0,4294967295]) (a.k.a the number of distinct keys being the most significant 32~bits of elements, leading to a two zero bytes padding).
```

Comment:
```suggestion
- Write as ``long``/``uint64`` the distinct number of buckets (restricted to integers in [0,4294967295]) (a.k.a the number of distinct keys being the most significant 32~bits of elements, leading to a two zero bytes padding).
```
Why is storing a 32-bit number in a ``uint64`` 2 bytes of padding and not 4, and why are we doing it at all instead of just storing 4 bytes here?

Reply:
> why are we doing it at all instead of just storing 4 bytes here?

This has been discussed in https://github.com/RoaringBitmap/RoaringFormatSpec/pull/12#pullrequestreview-1179395091. The only strong reason is that CRoaring has seemingly a greater stock of serialized 64-bit Roaring bitmaps, then I felt preferable converging to their format.

> Why is storing a 32-bit number in a uint64 2 bytes of padding and not 4

I did not process the suggestion this was inducing a 2 bytes padding, but you are right this format induces a 4 bytes padding.

---

I would be keen discussing this with a CRoaring 64-bit implementation author to get if there is a real motivation or not.

if there is no real motivation, do we want to stick to CRoaring format, or produce yet another one ? Any way, I feel this format is improvable feature-wise a it does not enable trivial memory-mapping (as this would require writing ahead the (cumulative?) length of each bucket).

Reply:
Well, since it seems the current format is already established in some way by CRoaring, I suppose we gotta enshrine this one for now, and if we want a memory mappable one later, we define a new format.